File: C++(3)/6.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string cadena;
    cout << "Ingrese una cadena de caracteres: ";
    cin >> cadena;

    int n = cadena.size();

    for (int i = 0; i < n-1; i++) {
        for (int j = 0; j < n-i-1; j++) {
            if (cadena[j] > cadena[j+1]) {
                char temp = cadena[j];
                cadena[j] = cadena[j+1];
                cadena[j+1] = temp;
            }
        }
    }

    cout << "La cadena ordenada es: " << cadena << endl;
    return 0;
}

File: Py (3)/6.py
def burbuja_cadena(cadena):
    arreglo = list(cadena)
    n = len(arreglo)

    for i in range(n-1):
        for j in range(n-i-1):
            if arreglo[j] > arreglo[j+1]:
                arreglo[j], arreglo[j+1] = arreglo[j+1], arreglo[j]

    return "".join(arreglo)

cadena = input("Ingrese una cadena de caracteres: ")
print("La cadena ordenada es:", burbuja_cadena(cadena))
